docs/architecture.md

```diff
@@ -0,0 +1,21 @@
++++
+title = "Architecture"
++++
+
+## Architecture
+
+### Roaring bitmap storage format
+
+Bitmaps are persisted to disk using a file format very similar to the [Roaring Bitmap format spec](https://github.com/RoaringBitmap/RoaringFormatSpec). Pilosa's format uses 64-bit IDs, so it is not binary-compatible with the spec. Some parts of the format are simpler, and an additional section is included. Specific differences include:
+
+* The cookie is always bytes 0-3; the container count is always bytes 4-7, never bytes 2-3.
+* The cookie includes file format version in bytes 2-3 (currently equal to zero).
+* The offset header section is always included.
+* RLE runs are serialized as [start, last], not [start, length].
```

Comment:
if it was something public facing i might want to think about it more, but since it's pretty deep in the internals, i don't think it's worth changing.